main.rb

```diff
@@ -0,0 +1,93 @@
+
+module Enumerable
+    def my_each
+        for item in self
+            yield(item)
+        end
+    end
+
+  def my_each_with_index
+      for item_index in (0...self.size)
+            yield(self[item_index], item_index)
+      end
+    end
+
+  def my_select
+        new_items = []
+        my_each do |item|
+          new_items.push(yield(item))
+        end
+        new_items
+  end
```

Comment:
- [ ] should have working `my_select` identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- Your method is not returning an enumerator when no block is given.
- Your method is returning an array with true or false instead of the actual elements that match the given argument. Please fix this.